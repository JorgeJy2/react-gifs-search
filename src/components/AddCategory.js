import React,{useState} from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');
    
    const handleInputChange = ({target: {value}})=> {
        // console.log(value);
        setinputValue(value)
    };


    const handleSubmit = (e)=> {
       e.preventDefault();
    //    console.log('Handle submit');
       
       if(inputValue.trim().length > 2){
        //    console.log(e);

           setCategories(categories =>  [inputValue,...categories]);
           setinputValue('');
       }
    };


    return (
        <form onSubmit={handleSubmit}>
            {/* <p> Add Category</p> */}
            <input 
                placeholder="Search"
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
