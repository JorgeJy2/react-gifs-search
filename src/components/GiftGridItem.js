
import React from 'react'
import PropTypes from 'prop-types';

export const GiftGridItem = ( {url,title} ) => {
    return (
        <div className="card animate__animated animate__fadeInDown">
           <img src={url} alt={title}/>
           <p>{title}</p>
        </div>
    )
}

GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

