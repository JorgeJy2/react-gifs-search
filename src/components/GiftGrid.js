import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from './hooks/UseFetchGifs';
import { GiftGridItem } from './GiftGridItem';

export const  GiftGrid = ({category}) => {

    const {data: images,loading} = useFetchGifs(category);   
    
    return (
        <>
        <p className="category">{category}</p>
        {loading && <p id="message-loading">Loading....</p>}
        <div className="card-grid">
                {
                    images?.map((img) => (
                        <GiftGridItem 
                            key={img.id}
                            {...img}
                            />
                    ))
                }
        </div>
        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
};
