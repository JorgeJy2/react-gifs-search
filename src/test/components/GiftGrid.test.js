import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GiftGrid } from '../../components/GiftGrid';
import { useFetchGifs } from '../../components/hooks/UseFetchGifs';

jest.mock('../../components/hooks/UseFetchGifs');

describe('test GiftGridI', ()=> {
    const category = 'dogs';
    test('test snapshot', ()=> {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( < GiftGrid category={category} / > );

        expect(wrapper).toMatchSnapshot();
    });

    test('should view items when load data.', ()=> {
        const gifs = [
            {
                id: 'abc12',
                url: 'https://localhost/img',
                title: 'test'
            },
            {
                id: 'abc12',
                url: 'https://localhost/img',
                title: 'test'
            }
        ]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow( < GiftGrid category={category} / > );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("#message-loading").exists()).toBe( false );
        expect(wrapper.find("GiftGridItem").length).toBe(gifs.length);
    });
});