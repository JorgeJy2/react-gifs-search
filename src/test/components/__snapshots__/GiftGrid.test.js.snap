// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GiftGridI should view items when load data. 1`] = `
<Fragment>
  <p
    className="category"
  >
    dogs
  </p>
  <div
    className="card-grid"
  >
    <GiftGridItem
      id="abc12"
      key="abc12"
      title="test"
      url="https://localhost/img"
    />
    <GiftGridItem
      id="abc12"
      key="abc12"
      title="test"
      url="https://localhost/img"
    />
  </div>
</Fragment>
`;

exports[`test GiftGridI test snapshot 1`] = `
<Fragment>
  <p
    className="category"
  >
    dogs
  </p>
  <p
    id="message-loading"
  >
    Loading....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
